storage/retention.go

```diff
@@ -121,12 +121,10 @@ func (s *retentionEnforcer) expireData(ctx context.Context, buckets []*influxdb.
 	defer logEnd()
 
 	// Snapshot to clear the cache to reduce write contention.
-	span, _ := tracing.StartSpanFromContextWithOperationName(ctx, "retention_snapshot")
-	if err := s.Snapshotter.WriteSnapshot(context.Background()); err != nil {
-		if err != tsm1.ErrSnapshotInProgress {
-			logger.Warn("unable to snapshot cache before retention", zap.Error(err))
-			tracing.LogError(span, err)
-		}
+	span, _ := tracing.StartSpanFromContext(ctx)
```

Comment:
Delete this line. `WriteSnapshot()` (now `writeSnapshot()`) creates it's own span, inheriting from the parent in `ctx`